File: test/api/issue4889.cpp
/*********************                                                        */
/*! \file issue4889.cpp
 ** \verbatim
 ** Top contributors (to current version):
 **   Andrew Reynolds
 ** This file is part of the CVC4 project.
 ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
 ** in the top-level source directory and their institutional affiliations.
 ** All rights reserved.  See the file COPYING in the top-level source
 ** directory for licensing information.\endverbatim
 **
 ** \brief Test for issue #4889
 **/

#include "api/cvc4cpp.h"

using namespace CVC4::api;

int main()
{
#ifdef CVC4_USE_SYMFPU
  Solver slv;
  Sort sort_int = slv.getIntegerSort();
  Sort sort_array = slv.mkArraySort(sort_int, sort_int);
  Sort sort_fp128 = slv.mkFloatingPointSort(15, 113);
  Sort sort_fp32 = slv.mkFloatingPointSort(8, 24);
  Sort sort_bool = slv.getBooleanSort();
  Term const0 = slv.mkConst(sort_fp32, "_c0");
  Term const1 = slv.mkConst(sort_fp32, "_c2");
  Term const2 = slv.mkConst(sort_bool, "_c4");
  Term ite = slv.mkTerm(ITE, const2, const1, const0);
  Term rem = slv.mkTerm(FLOATINGPOINT_REM, ite, const1);
  Term isnan = slv.mkTerm(FLOATINGPOINT_ISNAN, rem);
  slv.checkSatAssuming(isnan);
#endif
  return 0;
}

File: test/api/python/issue4889.py
###############################################################################
# Top contributors (to current version):
#   Yoni Zohar, Alex Ozdemir, Aina Niemetz
#
# This file is part of the cvc5 project.
#
# Copyright (c) 2009-2024 by the authors listed in the file AUTHORS
# in the top-level source directory and their institutional affiliations.
# All rights reserved.  See the file COPYING in the top-level source
# directory for licensing information.
# #############################################################################
#
# Test for issue #4889
##

import cvc5
from cvc5 import Kind

tm = cvc5.TermManager()
slv = cvc5.Solver(tm)
sort_int = tm.getIntegerSort()
sort_array = tm.mkArraySort(sort_int, sort_int)
sort_fp128 = tm.mkFloatingPointSort(15, 113)
sort_fp32 = tm.mkFloatingPointSort(8, 24)
sort_bool = tm.getBooleanSort()
const0 = tm.mkConst(sort_fp32, "_c0")
const1 = tm.mkConst(sort_fp32, "_c2")
const2 = tm.mkConst(sort_bool, "_c4")
ite = tm.mkTerm(Kind.ITE, const2, const1, const0)
rem = tm.mkTerm(Kind.FLOATINGPOINT_REM, ite, const1)
isnan = tm.mkTerm(Kind.FLOATINGPOINT_IS_NAN, rem)
slv.checkSatAssuming(isnan)
